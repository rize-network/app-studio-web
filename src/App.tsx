import React, { Suspense } from 'react';
import { Route, Routes, useLocation } from 'react-router-dom';
import { ComponentRouter } from 'src/routes/routes';
import { View, Text } from 'app-studio';

export const App: React.FC = () => {
  const location = useLocation();
  return (
    <Suspense
      fallback={
        <View
          height="100vh"
          display="flex"
          alignItems="center"
          justifyContent="center"
        >
          <Text>Loading...</Text>
        </View>
      }
    >
      <Routes location={location}>
        {ComponentRouter.map((item, index) => (
          <Route key={index} path={item.path} element={item.element} />
        ))}
      </Routes>
    </Suspense>
  );
};

export default App;
