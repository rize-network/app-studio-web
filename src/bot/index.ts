import { Bot } from './Bot';
import { DocuCode } from './DocuCode';

async function main() {
  const projectName = 'MDX Documentation';
  const descriptionPath = 'docs/description.md';

  // Get componentName from command line arguments
  const componentName = process.argv[2]; // Assuming it's the first argument passed
  if (!componentName) {
    console.error('Please provide a component name as an argument.');
    process.exit(1);
  }

  console.log(`Generating documentation for component: ${componentName}`);

  const basePath = 'src/components';
  const propsPath = `src/data/props/${componentName}.props.json`;
  const componentFolder = `${basePath}/${componentName}`;
  const componentPath = `${basePath}/${componentName}/${componentName}`;

  const assistantGPT = new Bot(componentName);

  // Implementation for Documentation of the code
  console.log('Initializing documentation process...');
  const docuCode = new DocuCode(componentName);
  const assistantDocumentation = await assistantGPT.init(
    projectName,
    [],
    descriptionPath,
    'docu'
  );

  await docuCode.processDirectory(componentFolder, assistantDocumentation.id);

  // Generating props file for code
  console.log('Generating props file...');
  const { fileIds } = await assistantGPT.addFiles(componentPath);

  const assistantCreation = await assistantGPT.init(
    projectName,
    fileIds,
    descriptionPath,
    'props'
  );

  await assistantGPT.response(assistantCreation.id, propsPath);

  await assistantGPT.MarkdownGeneration(
    componentFolder,
    componentName,
    propsPath
  );

  console.log('Documentation generation completed.');
}

main().catch((error) => {
  console.error('Error during the execution of main:', error);
});
