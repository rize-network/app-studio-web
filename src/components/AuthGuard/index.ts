export { AuthGuard, withAuthGuard, RouteAuthGuard, AuthStatus } from './AuthGuard';
