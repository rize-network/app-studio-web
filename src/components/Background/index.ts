export { Background } from './Background';
export type {
  BackgroundProps,
  AuroraBackgroundProps,
  MeteorsProps,
  WallProps,
  ParticlesProps,
  GridProps,
  RipplesProps,
  BackgroundImageProps,
  BackgroundVideoProps,
  BackgroundGradientProps,
  BackgroundStyles,
} from './Background/Background.props';
export type {
  BackgroundEffect,
  AuroraDirection,
  MeteorTiming,
  BorderAnimationStyles,
  StrokeAnimationStyles,
  BackgroundContextType,
  BackgroundSize,
  BackgroundPosition,
  BackgroundRepeat,
  BlendMode,
} from './Background/Background.type';
