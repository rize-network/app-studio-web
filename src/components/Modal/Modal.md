### **Import**

```tsx static
import { Modal } from 'app-studio';
```

### **Default**

It uses the isOpen and onClose properties to display the modal.

```tsx
import { Button } from '../Button/Button';
import { useState } from 'react';

const [show, setShow] = useState(false);

<>
  <Button onClick={() => setShow(true)} isAuto>
    Open Modal
  </Button>
  <Modal.Overlay isOpen={show} onClose={() => setShow(false)}>
    <Modal.Container>
      <Modal.Header>Title</Modal.Header>
      <Modal.Body>
        Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget quam.
        Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Praesent commodo cursus magna, vel scelerisque
        nisl consectetur et.
      </Modal.Body>
      <Modal.Footer>
        <Button onClick={() => setShow(false)}>Cancel</Button>
      </Modal.Footer>
    </Modal.Container>
  </Modal.Overlay>
</>;
```

## **Modal Overlay**

It represents the overlay or backdrop that appears behind the modal content when the modal is displayed.

### **FullScreen**

“**is*fullScreen***” property changes the width and height of the modal so that it fits the entire screen.

```tsx
import { Button } from '../Button/Button';
import { useState } from 'react';

const [show, setShow] = useState(false);

<>
  <Button onClick={() => setShow(true)} isAuto>
    Full Screen
  </Button>
  <Modal.Overlay isOpen={show} onClose={() => setShow(false)}>
    <Modal.Container isFullScreen>
      <Modal.Header>Title</Modal.Header>
      <Modal.Body>
        Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget quam.
        Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Praesent commodo cursus magna, vel scelerisque
        nisl consectetur et.
      </Modal.Body>
      <Modal.Footer>
        <Button onClick={() => setShow(false)}>Cancel</Button>
      </Modal.Footer>
    </Modal.Container>
  </Modal.Overlay>
</>;
```

### **Position**

“**_position_**” places the modal in the specified direction. It has a default positioning of “center”.

```tsx
import { useState } from 'react';
import { Button } from '../Button/Button';
import { Horizontal } from '../Layout/Horizontal/Horizontal';

const [showTop, setShowTop] = useState(false);
const [showBottom, setShowBottom] = useState(false);
const [showRight, setShowRight] = useState(false);
const [showLeft, setShowLeft] = useState(false);
const [showCenter, setShowCenter] = useState(false);

<>
  <Horizontal gap={10} wrap="nowrap">
    <Button onClick={() => setShowTop(true)}>Top</Button>
    <Button onClick={() => setShowBottom(true)}>Bottom</Button>
    <Button onClick={() => setShowRight(true)}>Right</Button>
    <Button onClick={() => setShowLeft(true)}>Left</Button>
    <Button onClick={() => setShowCenter(true)}>Center</Button>
  </Horizontal>

  {showBottom && (
    <Modal.Overlay isOpen={showBottom} onClose={() => setShowBottom(false)} position="bottom">
      <Modal.Container>
        <Modal.Header>Title</Modal.Header>
        <Modal.Body>
          Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget quam.
          Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Praesent commodo cursus magna, vel scelerisque
          nisl consectetur et.
        </Modal.Body>
        <Modal.Footer>
          <Button onClick={() => setShowBottom(false)}>Cancel</Button>
        </Modal.Footer>
      </Modal.Container>
    </Modal.Overlay>
  )}
  {showTop && (
    <Modal.Overlay isOpen={showTop} onClose={() => setShowTop(false)} position="top">
      <Modal.Container>
        <Modal.Header>Title</Modal.Header>
        <Modal.Body>
          Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget quam.
          Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Praesent commodo cursus magna, vel scelerisque
          nisl consectetur et.
        </Modal.Body>
        <Modal.Footer>
          <Button onClick={() => setShowTop(false)}>Cancel</Button>
        </Modal.Footer>
      </Modal.Container>
    </Modal.Overlay>
  )}
  {showLeft && (
    <Modal.Overlay isOpen={showLeft} onClose={() => setShowLeft(false)} position="left">
      <Modal.Container>
        <Modal.Header>Title</Modal.Header>
        <Modal.Body>
          Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget quam.
          Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Praesent commodo cursus magna, vel scelerisque
          nisl consectetur et.
        </Modal.Body>
        <Modal.Footer>
          <Button onClick={() => setShowLeft(false)}>Cancel</Button>
        </Modal.Footer>
      </Modal.Container>
    </Modal.Overlay>
  )}
  {showRight && (
    <Modal.Overlay isOpen={showRight} onClose={() => setShowRight(false)} position="right">
      <Modal.Container>
        <Modal.Header>Title</Modal.Header>
        <Modal.Body>
          Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget quam.
          Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Praesent commodo cursus magna, vel scelerisque
          nisl consectetur et.
        </Modal.Body>
        <Modal.Footer>
          <Button onClick={() => setShowRight(false)}>Cancel</Button>
        </Modal.Footer>
      </Modal.Container>
    </Modal.Overlay>
  )}
  {showCenter && (
    <Modal.Overlay isOpen={showCenter} onClose={() => setShowCenter(false)} position="center">
      <Modal.Container>
        <Modal.Header>Title</Modal.Header>
        <Modal.Body>
          Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget quam.
          Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Praesent commodo cursus magna, vel scelerisque
          nisl consectetur et.
        </Modal.Body>
        <Modal.Footer>
          <Button onClick={() => setShowCenter(false)}>Cancel</Button>
        </Modal.Footer>
      </Modal.Container>
    </Modal.Overlay>
  )}
</>;
```

### **Blur**

“**_blur_**” adds a blur effect on the modal overlay.

```tsx
import { useState } from 'react';
import { Button } from '../Button/Button';

const [show, setShow] = useState(false);

<>
  <Button onClick={() => setShow(true)}>Blur Overlay</Button>
  <Modal.Overlay isOpen={show} onClose={() => setShow(false)} blur={10}>
    <Modal.Container>
      <Modal.Header>Title</Modal.Header>
      <Modal.Body>
        Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget quam.
        Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Praesent commodo cursus magna, vel scelerisque
        nisl consectetur et.
      </Modal.Body>
      <Modal.Footer>
        <Button onClick={() => setShow(false)}>Cancel</Button>
      </Modal.Footer>
    </Modal.Container>
  </Modal.Overlay>
</>;
```

### **Prevent Close**

“**_isClosePrevented_**” prevents the user from closing the modal when clicking outside of it.

```tsx
import { useState } from 'react';
import { Button } from '../Button/Button';

const [show, setShow] = useState(false);

<>
  <Button onClick={() => setShow(true)} isAuto>
    Prevent Close on Overlay
  </Button>
  <Modal.Overlay isOpen={show} onClose={() => setShow(false)} isClosePrevented>
    <Modal.Container>
      <Modal.Header>Title</Modal.Header>
      <Modal.Body>
        Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget quam.
        Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Praesent commodo cursus magna, vel scelerisque
        nisl consectetur et.
      </Modal.Body>
      <Modal.Footer>
        <Button onClick={() => setShow(false)}>Cancel</Button>
      </Modal.Footer>
    </Modal.Container>
  </Modal.Overlay>
</>;
```

## **Modal Container**

It represents the main container or wrapper that holds all the content and components of a modal.

### **Shapes**

“**_shape_**” gives a sharp or rounded edges to the modal container.

```tsx
import { useState } from 'react';
import { Button } from '../Button/Button';
import { Text } from '../Text/Text';
import { Horizontal } from '../Layout/Horizontal/Horizontal';

const [showSharp, setShowSharp] = useState(false);
const [showRounded, setShowRounded] = useState(false);

<>
  <Horizontal gap={10}>
    <Button onClick={() => setShowSharp(true)}>Sharp</Button>
    <Button onClick={() => setShowRounded(true)}>Rounded</Button>
  </Horizontal>

  {showSharp && (
    <Modal.Overlay isOpen={showSharp} onClose={() => setShowSharp(false)}>
      <Modal.Container variant="sharp">
        <Modal.Body>
          <Text isTruncated maxLines={2}>
            Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget
            quam. Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Praesent commodo cursus magna, vel
            scelerisque nisl consectetur et.
          </Text>
        </Modal.Body>
      </Modal.Container>
    </Modal.Overlay>
  )}
  {showRounded && (
    <Modal.Overlay isOpen={showRounded} onClose={() => setShowRounded(false)}>
      <Modal.Container variant="rounded">
        <Modal.Body>
          <Text isTruncated maxLines={2}>
            Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget
            quam. Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Praesent commodo cursus magna, vel
            scelerisque nisl consectetur et.
          </Text>
        </Modal.Body>
      </Modal.Container>
    </Modal.Overlay>
  )}
</>;
```

### **Shadow**

“**_shadow_**” adds a shadow effect on the modal container.

```tsx
import { useState } from 'react';
import { Button } from '../Button/Button';
import { Text } from '../Text/Text';

const [show, setShow] = useState(false);

<>
  <Button onClick={() => setShow(true)}>Shadow</Button>

  {show && (
    <Modal.Overlay isOpen={show} onClose={() => setShow(false)}>
      <Modal.Container shadow={{ boxShadow: '0px 2px 8px rgba(0, 0, 0, 0.6)' }}>
        <Modal.Body>
          <Text isTruncated maxLines={2}>
            Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget
            quam. Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Praesent commodo cursus magna, vel
            scelerisque nisl consectetur et.
          </Text>
        </Modal.Body>
      </Modal.Container>
    </Modal.Overlay>
  )}
</>;
```

## **Modal Header**

It represents the header section of a modal, which typically contains a title or heading and optionally other elements such as a close button or additional controls.

### **Close Button Alignment**

“**_buttonPosition_**” places the closing button found in the header modal in the specified direction. It has a default position of “**right**”.

```tsx
import { useState } from 'react';
import { Button } from '../Button/Button';
import { Horizontal } from '../Layout/Horizontal/Horizontal';

const [showRight, setShowRight] = useState(false);
const [showWithout, setShowWithout] = useState(false);
const [showLeft, setShowLeft] = useState(false);

<>
  <Horizontal gap={10}>
    <Button onClick={() => setShowRight(true)} isAuto>
      Close button on the right
    </Button>
    <Button onClick={() => setShowWithout(true)} isAuto>
      Without Close Button
    </Button>
    <Button onClick={() => setShowLeft(true)} isAuto>
      Without Close left
    </Button>
  </Horizontal>
  {showRight && (
    <Modal.Overlay isOpen={showRight} onClose={() => setShowRight(false)}>
      <Modal.Container>
        <Modal.Header> Close Button</Modal.Header>
        <Modal.Body>Cras mattis consectetur purus sit amet fermentum.</Modal.Body>
        <Modal.Footer>
          <Button onClick={() => setShowRight(false)}>Cancel</Button>
        </Modal.Footer>
      </Modal.Container>
    </Modal.Overlay>
  )}
  {showWithout && (
    <Modal.Overlay isOpen={showWithout} onClose={() => setShowWithout(false)}>
      <Modal.Container>
        <Modal.Header buttonPosition="none"> Close Button</Modal.Header>
        <Modal.Body>Cras mattis consectetur purus sit amet fermentum.</Modal.Body>
        <Modal.Footer>
          <Button onClick={() => setShowWithout(false)}>Cancel</Button>
        </Modal.Footer>
      </Modal.Container>
    </Modal.Overlay>
  )}
  {showLeft && (
    <Modal.Overlay isOpen={showLeft} onClose={() => setShowLeft(false)}>
      <Modal.Container>
        <Modal.Header buttonPosition="left">Close Button</Modal.Header>
        <Modal.Body>Cras mattis consectetur purus sit amet fermentum.</Modal.Body>
        <Modal.Footer>
          <Button onClick={() => setShowLeft(false)}>Cancel</Button>
        </Modal.Footer>
      </Modal.Container>
    </Modal.Overlay>
  )}
</>;
```

## **Modal Body**

It represents the main content area of a modal, where the primary information or user interaction elements are displayed.

### **LargeContent**

A vertical scroll is displayed, if the content height exceed the body height.

```tsx
import { useState } from 'react';
import { Button } from '../Button/Button';
import { Text } from '../Text/Text';

const [show, setShow] = useState(false);

<>
  <Button onClick={() => setShow(true)}>Scroll</Button>
  {show && (
    <Modal.Overlay isOpen={show} onClose={() => setShow(false)}>
      <Modal.Container>
        <Modal.Header buttonPosition="none">
          <Text size="lg" weight="semibold">
            Title
          </Text>
        </Modal.Header>
        <Modal.Body height="200px">
          Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis
          consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur
          purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit
          amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet Cras
          mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis
          consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur
          purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit
          amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet
          fermentum.Cras mattis consectetur purus sit amet fermentum.
        </Modal.Body>
        <Modal.Footer>
          <Button onClick={() => setShow(false)}>Cancel</Button>
        </Modal.Footer>
      </Modal.Container>
    </Modal.Overlay>
  )}
</>;
```

## **Modal Footer**

It represents the footer section of a modal, which typically contains buttons or controls for actions related to the modal's content.

```tsx
import { useState } from 'react';
import { Button } from '../Button/Button';
import { Text } from '../Text/Text';

const [show, setShow] = useState(false);

<>
  <Button onClick={() => setShow(true)}>Footer</Button>
  {show && (
    <Modal.Overlay isOpen={show} onClose={() => setShow(false)}>
      <Modal.Container>
        <Modal.Header buttonPosition="none">
          <Text size="lg" weight="semibold">
            Title
          </Text>
        </Modal.Header>
        <Modal.Body height="200px">
          Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis
          consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur
          purus sit amet fermentum.
        </Modal.Body>
        <Modal.Footer>
          <Button onClick={() => setShow(false)}>Cancel</Button>
        </Modal.Footer>
      </Modal.Container>
    </Modal.Overlay>
  )}
</>;
```

## **Types**

```tsx static
type Position = 'top' | 'left' | 'right' | 'bottom' | 'center';
```

```tsx static
type Shape = 'sharp' | 'rounded';
```

```tsx static
type CloseButtonPosition = 'left' | 'right' | 'none';
```

```tsx static
type Size = 'xs' | 'sm' | 'md' | 'lg' | 'xl';
```

## **Default Styles**

#### **OverlayAlignments**

```tsx static
const OverlayAlignments: Record<Position, CSSProperties> = {
  center: { justifyContent: 'center', alignItems: 'center' },
  top: { justifyContent: 'center' },
  right: { justifyContent: 'flex-end', alignItems: 'center' },
  bottom: { justifyContent: 'center', alignItems: 'flex-end' },
  left: { alignItems: 'center' },
};
```

#### **ContainerShapes**

```tsx static
const ContainerShapes: Record<Shape, CSSProperties> = {
  sharp: { borderRadius: 0 },
  rounded: { borderRadius: 4 },
};
```

#### **HeaderIconSizes**

```tsx static
const HeaderIconSizes: Record<Size, number> = {
  xs: 12,
  sm: 16,
  md: 20,
  lg: 24,
  xl: 28,
};
```
