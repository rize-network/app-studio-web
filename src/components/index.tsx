export * from './Button/Button';
export * from './Form/Checkbox/Checkbox';
export * from './Form/CountryPicker/CountryPicker';
export * from './Form/DatePicker/DatePicker';
export * from './Form/Password/Password';
export * from './Form/Select/Select';
export * from './Form/Switch/Switch';
export * from './Form/Switch/Switch';
export * from './Form/TextArea/TextArea';
export * from './Form/TextField/TextField';
export * from './Layout/Center/Center';
export * from './Formik';
export * from './Layout/Horizontal/Horizontal';
export * from './Layout/Vertical/Vertical';
export * from './Layout/View/View';
export * from './Link/Link';
export * from './Loader/Loader';
export * from './Modal/Modal';
export * from './Text/Text';
export * from './File/File';
export * from './Tabs/Tabs';
export * from './Alert/Alert';
export * from './AspectRatio/AspectRatio';
export * from './Avatar/Avatar';
export * from './Badge/Badge';
export * from './Toggle/Toggle';
export * from './ToggleGroup/ToggleGroup';
export * from './Message/Message';
