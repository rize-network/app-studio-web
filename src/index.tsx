import React from 'react';
import ReactDOM from 'react-dom/client';
import AppProvider from 'src/providers';
import { App } from './App';
import { ComboBoxProvider } from './components/Form/ComboBox/ComboBox/ComboBox.provider';

const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement
);
root.render(
  <React.StrictMode>
    <AppProvider>
      <ComboBoxProvider>
        <App />
      </ComboBoxProvider>
    </AppProvider>
  </React.StrictMode>
);
