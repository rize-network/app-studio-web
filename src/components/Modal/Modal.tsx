import React from 'react';

import { ModalType } from './Modal/Modal.props';
import {
  ModalBody,
  ModalContainer,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
} from './Modal/Modal.view';
import { ModalLayout } from './Modal/Modal.layout';

/**
 * It is a custom content overlay that appears on top of the main screen.
 */
// eslint-disable-next-line react/prop-types
export const Modal: ModalType = ({ children }) => <>{children}</>;

Modal.Overlay = ModalOverlay;
Modal.Container = ModalContainer;
Modal.Header = ModalHeader;
Modal.Body = ModalBody;
Modal.Footer = ModalFooter;
Modal.Layout = ModalLayout;

export * from './Modal/Modal.store';
