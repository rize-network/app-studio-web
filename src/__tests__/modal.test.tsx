import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from 'src/components';
import Modal from 'src/components/Modal/Modal';

test('Modal to match snapshot', () => {
  const tree = renderer
    .create(
      <Modal.Overlay isOpen={true} onClose={() => {}} position="left" blur={10} isFullScreen>
        <Modal.Container variant="sharp" shadow={{ boxShadow: 'none' }} isClosePrevented>
          <Modal.Header buttonSize="xl" buttonPosition="left">
            Modal Container
          </Modal.Header>
          <Modal.Body height="200px">
            Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras
            mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis
            consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis
            consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis
            consectetur purus sit amet Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus
            sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet
            fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet
            fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet
            fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.
            Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras
            mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis
            consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis
            consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis
            consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.
          </Modal.Body>
          <Modal.Footer>
            <Button>Cancel</Button>
          </Modal.Footer>
        </Modal.Container>
      </Modal.Overlay>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
