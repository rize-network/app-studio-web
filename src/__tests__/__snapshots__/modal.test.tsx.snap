// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal to match snapshot 1`] = `
<div
  className="sc-bdfCDU ePQxSx"
  display="flex"
  height="100%"
  onClick={[Function]}
  visibility="visible"
  width="100%"
>
  <div
    className="sc-bdfCDU hRvVii"
    display="flex"
    height="100vh"
    onClick={[Function]}
    width="100vw"
  >
    <div
      className="sc-bdfCDU gdfEff"
      cursor="default"
      display="flex"
      height="fit-content"
      onClick={[Function]}
      overflow="hidden"
      width={600}
    >
      <div
        className="sc-bdfCDU MMgKI"
        display="flex"
      >
        <button
          className="sc-bdfCDU dXOPYQ"
          color="color.white"
          cursor="pointer"
          disabled={false}
          display="flex"
          filter="none"
          
          fontWeight={600}
          letterSpacing={1.25}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          transform=""
          width="fit-content"
        >
          <div
            className="sc-bdfCDU dpBfNq"
            display="flex"
          >
            <div
              className="sc-bdfCDU cccTHv"
              display="flex"
              height="20px"
              width="20px"
            >
              <svg
                fill="#000000"
                height="20px"
                viewBox="0 0 1024 1024"
                width="20px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  id="SVGRepo_bgCarrier"
                  strokeWidth="0"
                />
                <g
                  id="SVGRepo_tracerCarrier"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
                <g
                  id="SVGRepo_iconCarrier"
                >
                  <path
                    d="M764.288 214.592 512 466.88 259.712 214.592a31.936 31.936 0 0 0-45.12 45.12L466.752 512 214.528 764.224a31.936 31.936 0 1 0 45.12 45.184L512 557.184l252.288 252.288a31.936 31.936 0 0 0 45.12-45.12L557.12 512.064l252.288-252.352a31.936 31.936 0 1 0-45.12-45.184z"
                    fill="theme.primary"
                  />
                </g>
              </svg>
            </div>
          </div>
        </button>
        Modal Container
      </div>
      <div
        className="sc-bdfCDU csgvYH"
        height="200px"
      >
        Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum. Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.Cras mattis consectetur purus sit amet fermentum.
      </div>
      <div
        className="sc-bdfCDU jmYYGs"
        display="flex"
      >
        <button
          className="sc-bdfCDU bjTbpr"
          color="color.white"
          cursor="pointer"
          disabled={false}
          display="flex"
          filter="brightness(1)"
          
          fontWeight={600}
          letterSpacing={1.25}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          transform=""
          width={144}
        >
          <div
            className="sc-bdfCDU dpBfNq"
            display="flex"
          >
            Cancel
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
